sdk/identity/Azure.Identity/src/OnBehalfOfCredential.cs

```diff
@@ -0,0 +1,169 @@
+            string tenantId,
+            string clientId,
+            X509Certificate2 certificate,
+            string userAssertion,
+            OnBehalfOfCredentialOptions options,
+            CredentialPipeline pipeline,
+            MsalConfidentialClient client)
+            : this(
+                tenantId,
+                clientId,
+                new X509Certificate2FromObjectProvider(certificate ?? throw new ArgumentNullException(nameof(certificate))),
+                userAssertion,
+                options,
+                pipeline,
+                client)
+        { }
+
+        internal OnBehalfOfCredential(
+            string tenantId,
+            string clientId,
+            IX509Certificate2Provider certificateProvider,
+            string userAssertion,
+            OnBehalfOfCredentialOptions options,
+            CredentialPipeline pipeline,
+            MsalConfidentialClient client)
+        {
+            _tenantId = Validations.ValidateTenantId(tenantId, nameof(tenantId));
+            _clientId = clientId ?? throw new ArgumentNullException(nameof(clientId));
+            _allowMultiTenantAuthentication = options?.AllowMultiTenantAuthentication ?? false;
+            _pipeline = pipeline ?? CredentialPipeline.GetInstance(options);
+            options ??= new OnBehalfOfCredentialOptions();
+            _userAssertion = new UserAssertion(userAssertion);
+
+            _client = client ??
+                      new MsalConfidentialClient(
+                          _pipeline,
+                          tenantId,
+                          clientId,
+                          certificateProvider,
+                          options.SendCertificateChain,
+                          options,
+                          options.RegionalAuthority,
+                          options.IsLoggingPIIEnabled);
+        }
+
+        /// <summary>
+        /// Creates an instance of the <see cref="OnBehalfOfCredential"/> with the details needed to authenticate with Azure Active Directory.
+        /// </summary>
+        /// <param name="tenantId">The Azure Active Directory tenant (directory) Id of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal</param>
+        /// <param name="clientSecret">A client secret that was generated for the App Registration used to authenticate the client.</param>
+        /// <param name="userAssertion">The access token that will be used by <see cref="OnBehalfOfCredential"/> as the user assertion when requesting On-Behalf-Of tokens.</param>
+        /// <param name="options">Options that allow to configure the management of the requests sent to the Azure Active Directory service.</param>
+        public OnBehalfOfCredential(
+            string tenantId,
+            string clientId,
+            string clientSecret,
+            string userAssertion,
+            OnBehalfOfCredentialOptions options = null)
```

Comment:
nit: why does this ctor use a default parameter where the ctor which takes an X509Cert has an explicit overload?